tiling.py

```diff
@@ -241,10 +249,17 @@ def __init__(self, height, *vertices):
         # Due to the dihedral symmetry, we know half the outer edges are just reflected copies of the other half
         # so we can delete them to get the set of edges whose normals form the basis
         self._translational_fd_vertices = [e[0] for e in self._translational_fd_edges]
+
+        #if only floor mirror, the tfd vertices should move downward height
+        if self._flag_ceiling:
+            for i in range(len(self._translational_fd_vertices)):
+                self._translational_fd_vertices[i] = self._translational_fd_vertices[i] - np.array((0, height, 0, 0))
+
         basis_edges = self._translational_fd_edges[0:len(self._translational_fd_edges)/2]
         self._translational_basis = \
             [2.0*(0.5 * (e[0] + e[1]) - self._vertices[ctr_vertex_index]) for e in basis_edges]
 
+        self._translational_basis.append(2 * height * np.array((0, 1, 0, 0)))
```

Comment:
Don't we only need to do this if there's a floor or ceiling mirror. Did you test with no floor and mirror?

Reply:
Yes, we need them. In translational_fundamental_domains(self), even vdiameter is 1, we still get a pos[2] * self._group.translational_subgroup_basis[2]) even though pos[2] is 0. We can put the check here, which means we also need to put the check in translational_fundamental_domains(self). So I just always calculate them. I tested with no floor and ceiling mirrors. It works.